docs/rules/prefer-number-properties.md

```diff
@@ -126,3 +126,39 @@ const isPositiveZero = value => value === 0 && 1 / value === Number.POSITIVE_INF
 // eslint unicorn/prefer-number-properties: ["error", {"checkInfinity": true}]
 const isNegativeZero = value => value === 0 && 1 / value === Number.NEGATIVE_INFINITY;
 ```
+
+### checkNaN
+
+Type: `boolean`\
+Default: `true`
+
+Pass `checkNaN: false` to disable check on `NaN`.
+
+#### Fail
+
+```js
+// eslint unicorn/prefer-number-properties: ["error", {"checkNaN": true}]
+const foo = NaN;
+```
+
+```js
+// eslint unicorn/prefer-number-properties: ["error", {"checkNaN": true}]
+const foo = -NaN;
+```
+
+#### Pass
+
+```js
+// eslint unicorn/prefer-number-properties: ["error", {"checkNaN": false}]
+const foo = NaN;
+```
+
+```js
+// eslint unicorn/prefer-number-properties: ["error", {"checkNaN": false}]
+const foo = -NaN;
+```
+
+```js
+// eslint unicorn/prefer-number-properties: ["error", {"checkNaN": true}]
```

Comment:
Add cases that specifically show that `Number.NaN` and `-Number.NaN` pass
I don't think this example is necessary.

Reply:
Oh sorry; I was confused because Inifinity doesn't have examples like that either